################################################################################
SHORT DESCRIPTION: 
################################################################################
Do not offload ip checksum to guests.

################################################################################
LONG DESCRIPTION: 
################################################################################
Disable NETIF_F_IP_CSUM_BIT & NETIF_F_IPV6_CSUM_BIT for xen-netdev.

################################################################################
CHANGELOG 
################################################################################
Original Author: unknown
Port to 3.18: Eric Chanudet <chanudete@ainfosec.com>

################################################################################
REMOVAL 
################################################################################
"feature-no-csum-offload" and "feature-ipv6-csum-offload" Xenstore node can be
used as well to enable the feature or not, this patch only forces it to off in
any case.

################################################################################
UPSTREAM PLAN
################################################################################
None, could be dealt at guest creation and be configurable.

################################################################################
INTERNAL DEPENDENCIES 
################################################################################
Unknown.

################################################################################
PATCHES 
################################################################################
Index: linux-3.18.16/drivers/net/xen-netback/interface.c
===================================================================
--- linux-3.18.16.orig/drivers/net/xen-netback/interface.c	2015-06-14 18:19:31.000000000 +0200
+++ linux-3.18.16/drivers/net/xen-netback/interface.c	2015-06-19 15:19:02.369645508 +0200
@@ -424,9 +424,7 @@
 	spin_lock_init(&vif->lock);
 
 	dev->netdev_ops	= &xenvif_netdev_ops;
-	dev->hw_features = NETIF_F_SG |
-		NETIF_F_IP_CSUM | NETIF_F_IPV6_CSUM |
-		NETIF_F_TSO | NETIF_F_TSO6;
+	dev->hw_features = NETIF_F_SG | NETIF_F_TSO | NETIF_F_TSO6;
 	dev->features = dev->hw_features | NETIF_F_RXCSUM;
 	dev->ethtool_ops = &xenvif_ethtool_ops;
 
Index: linux-3.18.16/drivers/net/xen-netfront.c
===================================================================
--- linux-3.18.16.orig/drivers/net/xen-netfront.c	2015-06-14 18:19:31.000000000 +0200
+++ linux-3.18.16/drivers/net/xen-netfront.c	2015-06-19 15:19:02.539643530 +0200
@@ -1335,11 +1335,8 @@
 
 	netdev->netdev_ops	= &xennet_netdev_ops;
 
-	netdev->features        = NETIF_F_IP_CSUM | NETIF_F_RXCSUM |
-				  NETIF_F_GSO_ROBUST;
-	netdev->hw_features	= NETIF_F_SG |
-				  NETIF_F_IPV6_CSUM |
-				  NETIF_F_TSO | NETIF_F_TSO6;
+	netdev->features        = NETIF_F_RXCSUM | NETIF_F_GSO_ROBUST;
+	netdev->hw_features	= NETIF_F_SG | NETIF_F_TSO | NETIF_F_TSO6;
 
 	/*
          * Assume that all hw features are available for now. This set
